server/vertx/src/main/scala/sttp/tapir/server/vertx/VertxEndpointOptions.scala

```diff
@@ -0,0 +1,47 @@
+package sttp.tapir.server.vertx
+
+import java.io.File
+
+import io.vertx.core.logging.{Logger, LoggerFactory}
+import sttp.tapir.server.{DecodeFailureHandler, LogRequestHandling, ServerDefaults}
+
+import scala.concurrent.ExecutionContext
+
+case class VertxEndpointOptions(
+  decodeFailureHandler: DecodeFailureHandler = ServerDefaults.decodeFailureHandler,
+  logger: Logger = LoggerFactory.getLogger("tapir-vertx"),
+  logRequestHandling: LogRequestHandling[Logger => Unit] = VertxEndpointOptions.defaultLogRequestHandling,
+  uploadDirectory: File = File.createTempFile("tapir", null).getParentFile.getAbsoluteFile,
+  private val specificExecutionContext: Option[ExecutionContext] = None
+) {
+
+  def executionContextOr(default: ExecutionContext): ExecutionContext =
+    specificExecutionContext.getOrElse(default)
+
+  def logWhenHandled(shouldLog: Boolean): VertxEndpointOptions =
```

Comment:
I think I'd either create forwarders for all `log*` methods, or drop these (again for consistency :) )

Reply:
That was my first thought, but in fact I think it could be a bit less clear for Vert.x users.
Vert.x has the notion of `HttpServerOptions`  which contains stuff like port, host, http(s), protocol version, etc.
Here, these options are more "endpoint" oriented (and btw, that's great that it's an implicit parameter : it allows for a default global value for every endpoint, but also allows to define options specificly for an endpoint).

So, I'm not sure whether  "ServerOptions" would be a good fit or nor. Especially, it allows the user to override the "execution context" (which is very important for Vert.x) endpoint per endpoint.

I can rename it, no worries, I'm just not sure if it's the right decision :)